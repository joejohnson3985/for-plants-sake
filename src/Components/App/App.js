import React from 'react';
import './App.scss';
import { Route, Switch, Redirect } from 'react-router-dom';
import Sidebar from '../Sidebar/Sidebar'
import Results from '../../Containers/Results/Results'
import Filter from '../../Containers/Filter/Filter'
import Header from '../../Containers/Header/Header'
import ExpandedTrail from '../ExpandedTrail/ExpandedTrail'
import ErrorDisplay from '../../Containers/ErrorDisplay/ErrorDisplay'
import Stats from '../../Containers/Stats/Stats'

const App = () => {
  return (
    <div className='content'>
      <Sidebar />
      <ErrorDisplay />
      <main>
        <Switch>
          <Redirect exact from="/" to="/search" />
          <Route exact path='/search' component={Filter} />
          <Route path='/my-hikes' component={Header} />
          <Route path='/trail/:id' component={ExpandedTrail} />
        </Switch>
        <Route exact path='/my-hikes/stats' component={Stats} />
        <Route exact path='/search' component={Results} />
        <Route path='/my-hikes' component={Results} />
      </main>
    </div>
  )
}

export default App;