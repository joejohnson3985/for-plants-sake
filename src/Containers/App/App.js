import React, { Component } from 'react';
import './App.scss';
import { Route } from 'react-router-dom';
import Sidebar from '../Sidebar/Sidebar'
import Results from '../Results/Results'
import Filter from '../Filter/Filter'

class App extends Component {

  render() {
    return(
      <div className='content'>
        <Sidebar />
        <main>
          <Route exact path='/' component={Filter} />
          <Results />
        </main>
      </div>
    )
  }
}


export default App;
