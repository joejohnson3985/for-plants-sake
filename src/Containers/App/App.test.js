import App from './App';
import { shallow } from 'enzyme';
import React from 'react';


const store = jest.fn()
let wrapper;

describe('App', () => {
  it('Should match the snapshot', () => {
    wrapper = shallow(
      <App />
    )
    expect(wrapper).toMatchSnapshot()
  })
}) 