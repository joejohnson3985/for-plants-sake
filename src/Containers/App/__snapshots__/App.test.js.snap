// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Should match the snapshot 1`] = `
<div
  className="content"
>
  <Sidebar />
  <Connect(ErrorDisplay) />
  <main>
    <Switch>
      <Redirect
        exact={true}
        from="/"
        to="/search"
      />
      <Route
        component={
          Object {
            "$$typeof": Symbol(react.memo),
            "WrappedComponent": [Function],
            "compare": null,
            "displayName": "Connect(Filter)",
            "type": [Function],
          }
        }
        exact={true}
        path="/search"
      />
      <Route
        component={
          Object {
            "$$typeof": Symbol(react.memo),
            "WrappedComponent": [Function],
            "compare": null,
            "displayName": "Connect(Header)",
            "type": [Function],
          }
        }
        path="/my-hikes"
      />
      <Route
        component={[Function]}
        path="/trail/:id"
      />
    </Switch>
    <Route
      component={
        Object {
          "$$typeof": Symbol(react.memo),
          "WrappedComponent": [Function],
          "compare": null,
          "displayName": "Connect(Results)",
          "type": [Function],
        }
      }
      exact={true}
      path="/search"
    />
    <Route
      component={
        Object {
          "$$typeof": Symbol(react.memo),
          "WrappedComponent": [Function],
          "compare": null,
          "displayName": "Connect(Results)",
          "type": [Function],
        }
      }
      path="/my-hikes"
    />
  </main>
</div>
`;
